Pre commit fixes

m-iwanicki opened the pull request.

m-iwanicki committed:
pre-commit: add pre-commit with default hooks + lychee

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
pre-commit: fix pre-commit failures

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

Commit:
LICENSE.md: format per markdownlint-fix

Commit:
docs/events.md: Remove the 'Video' bulletpoint

The video for Platform Security Summit 2019 was never published, so the
bulletpoint mentioning the video was removed.

Commit:
.pre-commit-config.yaml: remove shellcheck, disable lychee in CI

m-iwanicki commented:
lychee failures:

```
[docs/community.md]:
✗ [500] https://slack.osfw.dev/ | Failed: Network error: Internal Server Error

[docs/events.md]:
✗ [404] https://www.youtube.com/watch?v=A9GrlQsQc7Q?start=8152 | Failed: Network error: Not Found
✗ [429] https://vimeo.com/488140434 | Failed: Network error: Too Many Requests
```

Both vimeo and youtube work so only problem is `slack.osfw.dev`

m-iwanicki committed:
pre-commit: Add to requirements.txt. ignore .svg files

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
README.md: add short pre-commit information

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

krystian-hebel commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,67 @@
+      - id: yamllint
+        args: [-c=.yamllint]
+
+  - repo: https://github.com/3mdeb/hooks
```
I think this isn't necessary, this projects doesn't overlap with our product names.

m-iwanicki replied:
This hook checks more than just our product names. Currently it also checks Yocto/UEFI/NVIDIA and we will probably add more names/trademarks in the future.

krystian-hebel replied:
Is this check case-sensitive and does it skip URLs? If yes, I can approve adding it _after_ `TrenchBoot` is added to the list of names, otherwise I don't see how it helps.

m-iwanicki replied:
Ok removed it. This hook only checked if defined names/trademarks are correctly capitalized (e.g. UEFI which is used in this repo)
Maybe we will add it later when it's more mature.
47d18c03eba24459d5ad1e2ed1148bb5b9e26f8a

m-iwanicki commented:
In addition to previous lychee failures there is also:

* https://mirroronet.pl/pub/mirrors/FOSDEM/2020/K.4.601/firmware_itsoecs.mp4 - video can't be played becase the file is corrupt
* https://www.amd.com/system/files/TechDocs/24593.pdf - redirected, doesn't go directly to wanted document

m-iwanicki committed:
pre-commit: apply fixes after rebase

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

krystian-hebel commented on LICENSE.md:
```diff
@@ -1,159 +1,334 @@
+   Material:
 
-       A. retain the following if it is supplied by the Licensor with the Licensed Material:
+   i. identification of the creator(s) of the Licensed Material and any others
```
This and following points with roman numbering aren't properly indented.

m-iwanicki replied:
~~From what I can see they also weren't earlier~~. Nevermind, they were indented but it doesn't render correctly.
Markdown doesn't support roman numerals by default, I think it really depends how/where it's viewed. On GitHub at least in comments you can:

```
1. a
  1. a
  1. a
  1. a
1. a
```

which will result in:

1. a
    1. a
    1. a
    1. a
1. a

or add css in license document with custom styling.

```markdown
<style type="text/css">
    ol ol { list-style-type: upper-alpha; }
    ol ol ol { list-style-type: lower-roman; }
</style>
```

But still we would need to use automatic ordering (each list point would be `1.` in raw text)

m-iwanicki replied:
I added indent in 453b9ef336495333f86fe3fa5215a6b6595b7190 but it doesn't show when viewed on github.

krystian-hebel replied:
> But still we would need to use automatic ordering (each list point would be `1.` in raw text)

This isn't true:

```
1. a
   1. a
   2. a
   3. a
2. a
```

Renders as:

1. a
   1. a
   2. a
   3. a
2. a

Still, there is no way that I know of of using the roman numbering in source.

Perhaps renaming `LICENSE.md` to just `LICENSE` would solve some of the problems, but it may also create new ones. Let's leave it as is, it looks reasonably good in source form.

krystian-hebel commented on LICENSE.md:
```diff
@@ -1,159 +1,334 @@
+> avoidance of doubt, this paragraph does not form part of the public licenses.
 >
-> Creative Commons may be contacted at creativecommons.org
+> > Creative Commons may be contacted at creativecommons.org
```
Remove the second `>`.

m-iwanicki replied:
Removed in 05c60db0af2f029bb3258c1ef1c28a64d35a6d19

krystian-hebel commented on docs/CNAME:
```diff
@@ -1 +1 @@
-trenchboot.org
```
This isn't documented, but supposedly it won't work if there is newline character in the file. Please revert this change and add it as an exception, so it won't cause any delayed, hard to debug DNS problems.

m-iwanicki replied:
Done in 667dfbb024eedcf1e1e5f44d44a33e18c48b9ab7

krystian-hebel commented on docs/FAQ.md:
```diff
@@ -1,51 +1,51 @@
-from Microsoft Authority to Distribution/Installer/No Authority. The response
-why this is not of concern will be addressed in Question 2.
-
+For Linux systems doing both verified(secure) and measured boot, there
```
```suggestion
For Linux systems doing both verified (secure) and measured boot, there
```

m-iwanicki replied:
fixed 38411e46ed59e835bd55a8351bc3755265a420b3

krystian-hebel commented on docs/blueprints/Measured_Secure_Boot.md:
````diff
@@ -1,61 +1,65 @@
+respective blueprints. Finally, below is the execution flow for MSB
+for comparison with tboot's execution flow above.
+
+```shell
````
```suggestion
```text
```

m-iwanicki replied:
fixed 38411e46ed59e835bd55a8351bc3755265a420b3

krystian-hebel commented on docs/community.md:
```diff
@@ -1,6 +1,5 @@
-* [#trenchboot channel on OSFW Slack](https://slack.osfw.dev)
+- [TrenchBoot Google Group](https://groups.google.com/g/trenchboot-devel)
+- [Twitter](https://twitter.com/TrenchBoot)
+- [#trenchboot channel on OSFW Slack](https://slack.osfw.dev)
```
```suggestion
- [TrenchBoot room on Matrix](https://matrix.to/#/#OSFW-Trenchboot:matrix.org)
```

Slack channel is pretty much dead.

m-iwanicki replied:
fixed 38411e46ed59e835bd55a8351bc3755265a420b3

krystian-hebel commented on docs/steering-committee/.pages.yml:
```diff
@@ -1,2 +1,3 @@
+---
```
Please remove this file instead, it isn't used (I think).

m-iwanicki replied:
fixed 38411e46ed59e835bd55a8351bc3755265a420b3

krystian-hebel commented on docs/steering-committee/index.md:
```diff
@@ -1,6 +1,6 @@
 
-Here you will find the meeting minutes from the Steering Committiee and
-Community Meetings.
+Here you will find the meeting minutes from the Steering Committiee
```
```suggestion
Here you will find the meeting minutes from the Steering Committee
```

Are there any spellcheckers that can be added to the CI?

m-iwanicki replied:
Currently there is one, `codespell`, from what I can see it only works by looking for explicitly defined misspelled words (common misspellings). It suggested `Commitee` -> `Committee` change in the same file.
If you want to add/change to different spellchecker then maybe create new issue here or in https://github.com/3mdeb/hooks so we could decide on best one that can be used everywhere.

m-iwanicki replied:
fixed in 41cb961b084915a712daf5701954265d02f7bb14

krystian-hebel commented on docs/user-docs/test_matrix.md:
```diff
@@ -34,23 +36,13 @@ Devices and configurations on which TrenchBoot is known to not work:
 |:---------------------------------------------------------------------------------------------:|:----------:|:----------:|:-----:|
 | Supermicro M11SDV-8CT<br>(AMD EPYC 3000 Snowy Owl server)<br>(v0.4 by openQA)                 |  TPM 2.0   | 2019-today | CSM legacy boot,<br>unexpected PCR values |
 
-[aem]: https://github.com/TrenchBoot/qubes-antievilmaid
```
I think that keeping those close to where they are referenced make it easier to work with (modify, add new entries, move a part of document elsewhere etc.). Would it be possible to disable whatever check suggested this change?

m-iwanicki replied:
From what I can see currently nothing complains about that, so I'm not sure if @EduKav1813 wasn't overeager or maybe it was victim of another auto fix.

Changed it back here: 9611f9774549a0cd30d4c2338c5017f8dfef3c32

krystian-hebel commented on mkdocs.yml:
```diff
@@ -10,34 +11,34 @@ edit_uri: ""
 nav:
-    - Home: index.md
-    - Events:
+  - Home: index.md
```
First indent level is 2 spaces, and all of the following ones are 4, is this what lint suggested?

m-iwanicki replied:
I think it depends on config: https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.indentation

Currently from what I can see it wants 2 spaces between first character in word (that is not including `-`) in current line and first character (including `-`) in next line. In examples below I used dots for those 2 spaces:


*  ```yaml
    text:
    ..- text2
    ```

*  ```yaml
    - text:
      ..- text2
    ```

*  ```yaml
    text:
    ..text2
    ```

*  ```yaml
    - text:
      ..text2
    ```

krystian-hebel replied:
Interesting, definitely not what I expected, and difficult for text editors without knowledge of YAML syntax. Since this seems to be the standard way of formatting YAML files, I'm marking it as resolved, even though I still don't like it :slightly_smiling_face:

krystian-hebel commented on .markdownlint.yaml:
```diff
@@ -0,0 +1,196 @@
+#
+MD013:
+  # Number of characters
+  line_length: 80
```
All of the files were wrapped to lower number of characters per line, why is that?

m-iwanicki replied:
From what I can see markdownlint-fix doesn't wrap lines automatically so it's possible it was done manually or with other tool that was configured differently.

m-iwanicki committed:
.pre-commit-config: remove namespell

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
LICENSE.md: indent roman numerals, remove duplicated '>'

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
pre-commit: exclude CNAME from all checks, remove last newline

Quoting @krystian-hebel: "This isn't documented, but supposedly it
won't work if there is newline character in the file"

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
docs: fix dead link, fix formatting

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
docs: pre-commit: reapply fixes dropped during rebase

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

krystian-hebel commented on docs/blueprints/Linux_Late_Launch.md:
```diff
@@ -1,39 +1,38 @@
+* Verify measurement from LZ signature against CRTM
+* (AMD) Create/Setup TPM Event log
+* Extend hash of pubkey into PCR 18, record in event log
+* This aligns to TXT's DA scheme
```
This used to be on another indentation level.

m-iwanicki replied:
https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/a31855e7af2ad0366cb920fbc483a3e8e3b92a83

m-iwanicki replied:
force push: https://github.com/TrenchBoot/TrenchBoot.github.io/compare/a31855e7af2ad0366cb920fbc483a3e8e3b92a83..bd18e0db0237200a58a3d4bf0103b4ec187cfe7c

krystian-hebel commented on docs/blueprints/Linux_Late_Launch.md:
```diff
@@ -1,39 +1,38 @@
+* DEV/DMAR protect remainder of Linux kernel
+* Extend hash of Linux kernel into PCR 17, record in event log
+* Prepare and then handoff to Linux kernel proper
+* Need to support Legacy/CSM and UEFI environments
```
Same here.

m-iwanicki replied:
https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/a31855e7af2ad0366cb920fbc483a3e8e3b92a83

m-iwanicki replied:
force-push: https://github.com/TrenchBoot/TrenchBoot.github.io/compare/a31855e7af2ad0366cb920fbc483a3e8e3b92a83..bd18e0db0237200a58a3d4bf0103b4ec187cfe7c

krystian-hebel commented on docs/blueprints/Linux_Late_Launch.md:
```diff
@@ -1,39 +1,38 @@
+* Process system state passed up from LZ
+    - e.g. memory map, etc.
+* Setup Linux state for handover to mainline start up
+* Jump into mainline start up
```
Same here.

m-iwanicki replied:
https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/a31855e7af2ad0366cb920fbc483a3e8e3b92a83

krystian-hebel commented on docs/index.md:
```diff
@@ -4,6 +4,8 @@ hide:
   - toc
 ---
 
+# Home
```
This was deliberately removed from the front page some time ago, it doesn't look good. Please add `<!-- markdownlint-disable MD041 -->` instead.

m-iwanicki replied:
https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/a31855e7af2ad0366cb920fbc483a3e8e3b92a83

krystian-hebel commented on .markdownlint.yaml:
```diff
@@ -0,0 +1,196 @@
+# MD050/strong-style - Strong style should be consistent
+MD050:
+  # Strong style should be consistent
+  style: "asterisk"
```
Can we add [MD053](https://github.com/DavidAnson/markdownlint/blob/main/doc/md053.md)? Would it catch errors like the one fixed by https://github.com/TrenchBoot/TrenchBoot.github.io/pull/46/commits/bc3ffcff801aaafd24f1da5cd457ac8338cce9df?

m-iwanicki replied:
Yes:

```
markdownlint.............................................................Failed
- hook id: markdownlint
- exit code: 1

docs/blueprints/index.md:80:1 MD053/link-image-reference-definitions Link and image reference definitions should be needed [Duplicate link or image reference definition: "linux"] [Context: "[linux]: https://lore.kernel.o..."]
```

After changing:

```diff
-| Linux     | [In Progress][linux-up]         | N/A
+| Linux     | [In Progress][linux]         | N/A
 | Xen       | [Partially In Progress][xen-ap] | [In Progress][xen-qos]
 
 [grub-up]: https://lists.gnu.org/archive/html/grub-devel/2024-08/msg00088.html
 [grub-qos]: https://github.com/QubesOS/qubes-grub2/pull/13
-[linux-up]: https://lore.kernel.org/lkml/20240826223835.3928819-1-ross.philipson@oracle.com/
+[linux]: https://lore.kernel.org/lkml/20240826223835.3928819-1-ross.philipson@oracle.com/
```

Added in https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/6959805f5ba51e7b7aeb1a826b4e58ff2fe821d1

krystian-hebel reviewed (changes requested):
Few more changes, and a rebase will be needed because another PR was merged.

m-iwanicki committed:
pre-commit: add MD053 to .markdownlint.yaml

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
pre-commit: address review

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki commented:
@krystian-hebel rebased and added fixes. Is it correct now? Indent was hard to see (2 spaces only and wasn't rendered on GitHub correctly)
